app/views/sign_up_sheet/review_bids_others_work.html.erb

```diff
@@ -1,113 +1,103 @@
-<% elsif @assignment.num_reviews_allowed == @assignment.num_reviews_required %>
-    <h4>You should perform exactly <%= @assignment.num_reviews_allowed %> reviews </h4>
-<% else %>
-    <h4>You may perform between <%= @assignment.num_reviews_required %> and <%= @assignment.num_reviews_allowed %> reviews</h4>
+<% def display_assignment_reviews_info(assignment, num_reviews_completed) %>
+  <h2>Reviews for "<%= assignment.name %>"</h2>
+  <!-- add headers that provide information regarding review numbers -->
+  <% if assignment.num_reviews_allowed.nil? || assignment.num_reviews_allowed == -1 %>
+    <h4>Your instructor expects you to do <%= assignment.num_reviews_required %> reviews. You are not allowed to do any extra reviews. </h4>
+  <% elsif assignment.num_reviews_allowed == assignment.num_reviews_required %>
+    <h4>You should perform exactly <%= assignment.num_reviews_allowed %> reviews </h4>
+  <% else %>
+    <h4>You may perform between <%= assignment.num_reviews_required %> and <%= assignment.num_reviews_allowed %> reviews</h4>
+  <% end %>
+  <h4>You are required to do <%= assignment.num_reviews_required %> reviews</h4>
+  <h4>Number of reviews left: <%= assignment.num_reviews_allowed - num_reviews_completed %></h4>
 <% end %>
-<h4>You are required to do <%= @assignment.num_reviews_required %> reviews</h4>
-<h4> Number of reviews left: <%= @assignment.num_reviews_allowed - @num_reviews_completed %></h4>
 
-<!-- table with reviews available to complete -->
+
+######################################################################################################################################
+
+#This methods displays flash
+<% = flash_message %>
+  
+# This method displays the number of reviews to be done defined in review_bids_helper
+<%= display_assignment_reviews_info(@assignment, @num_reviews_completed) %>
+
+# -- table with reviews available to complete -->
 <table>
   <% review_no = 1 %>
   <% title = 'Review' %>
-  <!-- loop through each review map -->
+
   <% @review_mappings.each do |map| %>
-      <%@sorted_responses=[]%>
-      <% team = AssignmentTeam.find(map.reviewee_id) %>
-      <% participant = team.participants.first %>
+     # Create sorted_response array to store information from particular map id 
+    <% @sorted_responses = [] %>
+    <% team = AssignmentTeam.find(map.reviewee_id) %>
+    <% participant = team.participants.first %>
+
+    <% if participant %>
+      <% topic_id = SignedUpTeam.topic_id(participant.parent_id, participant.user_id) %>
+      <tr>
+        <%= display_topic_and_participant(title, review_no, topic_id, assignment, participant) %>
+        
+        <% latest_submission = team.most_recent_submission %>
+        <td>&nbsp;</td>
+
+        <% if !map.response.empty? && latest_submission %>
+          <% # Populate sorted response array from corresponding id and set array_not_empty flag %>
```

Comment:
This comment can be better. Populate the array with what?  What is the purpose of doing this?